### PR TITLE
catch exceptions when connecting to cluster & notify (#20)

### LINKED ISSUE #20: Display error if current cluster is not reachable
Currently if the current cluster isn't reachable nothing is displayed. We should display the cluster node and an error telling the user that it could not be reached:

image

### PR DESCRIPTION
**Pull request by adietish**
fixes #20

### CODE REVIEW

**Comment by adietish** [comment]
@jeffmaury, @sbouchet please review

steps: 
1. ASSERT: have a minikube cluster running
1. ASSERT: make sure minikube cluster is current-context (via ```kubectl config use-context <name>```)
1. EXEC: shut down minikube
1. EXEC: launch plugin

Result: 
Tree shows up with error as child to **Namespaces** & **Workloads/Pods**. 
image
You should also get an error notification and a entry in the "Event Log"
image

**Comment by adietish** [comment]
@jeffmaury , @sbouchet: What do you think of the notifications? Worth it? I also tend to think that the errors in the tree should be simplified text-wise (to be more like the popups). Thoughts?

**Comment by adietish** [comment]
improved the errors that are shown in the tree:
image
@jeffmaury, @sbouchet: Thoughts?

File: src/main/kotlin/org/jboss/tools/intellij/kubernetes/model/Cluster.kt [hunk 1]
```diff
@@ -120,7 +132,11 @@ open class Cluster(private val modelChange: IModelChangeObservable) : ICluster {
     }
 
     private fun loadAllNamespaces(): Sequence<Namespace> {
-        return  client.namespaces().list().items.asSequence()
+        try {
+            return client.namespaces().list().items.asSequence()
+        } catch (e: KubernetesClientException) {
+            throw KubernetesResourceException("Could not get all namespaces for sever ${client.masterUrl}", e)
```
Comment (jeffmaury) [inline src/main/kotlin/org/jboss/tools/intellij/kubernetes/model/Cluster.kt]: s/sever/server

File: src/main/kotlin/org/jboss/tools/intellij/kubernetes/tree/KubernetesTreeStructure.kt [hunk 2]
```diff
@@ -36,31 +40,30 @@ class KubernetesTreeStructure(private val project: Project) : AbstractTreeStruct
...
-        try {
             return when(element) {
                 rootElement ->
                     arrayOf(
                         Categories.NAMESPACES,
                         Categories.NODES,
                         Categories.WORKLOADS)
                 Categories.NAMESPACES ->
-                    getResourceModel().getAllNamespaces().sortedBy { it.metadata.name }.toTypedArray()
+                    runSafely("Loading resources failed", "Could not get all namespaces for server ${getResourceModel().getClient().masterUrl}") {
```
Comment (jeffmaury) [inline src/main/kotlin/org/jboss/tools/intellij/kubernetes/tree/KubernetesTreeStructure.kt]: Could notget namespaces to be coherent with the error for pod for example

File: src/main/kotlin/org/jboss/tools/intellij/kubernetes/utils/Notification.kt [hunk 3]
```diff
@@ -0,0 +1,23 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
```
Comment (sbouchet) [inline src/main/kotlin/org/jboss/tools/intellij/kubernetes/utils/Notification.kt]: should be 2020
```suggestion
 * Copyright (c) 2020 Red Hat, Inc.
```

**Comment by sbouchet** [comment]
@adietish : there is also notifications in Eclipse IDE, for odo explorer by example, but only for information purpose, like component creation/deletion and so on. Like @jeffmaury pointed out, errors should not be using notification, but proper logging.

**Comment by adietish** [comment]
@jeffmaury please re-review

File: src/main/kotlin/org/jboss/tools/intellij/kubernetes/utils/Notification.kt [hunk 4]
```diff
@@ -0,0 +1,23 @@
+/*******************************************************************************
```
Comment (jeffmaury) [inline src/main/kotlin/org/jboss/tools/intellij/kubernetes/utils/Notification.kt]: Is this class still used ?

Comment (adietish) [reply src/main/kotlin/org/jboss/tools/intellij/kubernetes/utils/Notification.kt]: No, it's not used any more. I thought about keeping it for future use. But then, non-used code is dead code, should be removed. I'll remove it.

**Pull request merged**